### ISSUE TITLE
7.0.2: Build scan does not work in tr_TR locale: No enum constant ANDROİD_PLUGİN_APPLİED

### ISSUE THREAD

**Issue by vlsi** [gradle/gradle#17361]
### Expected Behavior

Build should not fail.

### Current Behavior

```
* What went wrong:
com.gradle.scan.agent.serialization.scan.a.g (initialization failure)
> java.lang.IllegalArgumentException: No enum constant com.gradle.scan.agent.serialization.scan.a.d.ANDROİD_PLUGİN_APPLİED
```

### Steps to Reproduce 

`--scan -Duser.country=TR -Duser.language=tr build`

### Your Environment

Build scan URL: unable to produce build scan

**Comment by vlsi** [comment]
`--no-scan` does not work for some reason: 

```
% ./gradlew -Duser.language=tr -Duser.country=TR --no-scan jar -s

* What went wrong:
Could not initialize class com.gradle.scan.agent.serialization.scan.a.g

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Exception is:
java.lang.NoClassDefFoundError: Could not initialize class com.gradle.scan.agent.serialization.scan.a.g
        at com.gradle.scan.plugin.internal.g.a.a(SourceFile:22)
        at com.gradle.scan.plugin.internal.b.w.a.a(SourceFile:11)
        at com.gradle.scan.plugin.internal.b.c.a(SourceFile:38)
        at com.gradle.scan.plugin.internal.o.d.a(SourceFile:51)
        at com.gradle.scan.plugin.internal.q.a.b(SourceFile:111)
        at com.gradle.scan.plugin.internal.o.d.a(SourceFile:50)
        at com.gradle.scan.plugin.internal.o.b.a(SourceFile:23)
        at org.gradle.internal.enterprise.impl.DefaultGradleEnterprisePluginAdapter.buildFinished(DefaultGradleEnterprisePluginAdapter.java:77)
```

**Comment by marcphilipp** [comment]
This looks like `String.toUpperCase()` without an explicit locale is used somewhere:

```
jshell> "i".toUpperCase(Locale.forLanguageTag("tr-TR"));
$2 ==> "İ"
```

**Comment by paplorinc** [comment]
Thanks for reporting, I managed to reproduce it locally and provided a fix

**Comment by lptr** [comment]
This (i.e. the use of `toUpperCase()` without locale) is something where static analysis could help us.

**Comment by lptr** [comment]
@paplorinc is the fix in Gradle or in GE? Could you link to the PR please? If it's in GE, I think we can close this issue.

**Comment by paplorinc** [comment]
I can't link, it's a private repo, I'll close this when the PR is merged.

**Comment by paplorinc** [comment]
Merged, will be part of GE Gradle Plugin 3.6.3 and Maven Extension 1.10.1
